command/server.go

```diff
@@ -237,11 +239,29 @@ func (c *ServerCommand) Run(args []string) int {
 		DefaultLeaseTTL:    config.DefaultLeaseTTL,
 		ClusterName:        config.ClusterName,
 		CacheSize:          config.CacheSize,
+		PluginDirectory:    config.PluginDirectory,
 	}
 	if dev {
 		coreConfig.DevToken = devRootTokenID
 	}
 
+	if config.PluginDirectory == "" {
```

Comment:
Removed this altogether and put checks for an unset directory into the plugin catalog. Now the catalog won't accept new plugins if the directory isn't configured. There is no longer a default one.